src/target/target_kind.cc

```diff
@@ -257,6 +257,12 @@ TargetJSON TestTargetParser(TargetJSON target) {
 
 /**********  Register Target kinds and attributes  **********/
 
+#if defined(__arm__) || defined(__aarch64__)
+#define NATIVE_CPU "arm_cpu"
+#else
+#define NATIVE_CPU "cpu"
+#endif
```

Comment:
A general solution would definitely be preferable, do you have a suggestion of how one would use the target parser to do this?

Since we've gone from being x86 specfic to arm and x86 specfic I'd argue this is still an improvement over what we had before. We can always add support for more targets in future PRs.

It's worth noting that this has exposed a lot of false positive tests on the `arm_cpu` target, if the same thing happens for other targets it might be advantageous to do this iteratively in order to keep the diff contained.

Reply:
To make this more generic I wonder if we can use something like: https://github.com/apache/tvm/pull/14946/files#diff-f61b04b100f5145f2681340c81d3f2af221239594ed01e2e24896522329ce92cR2087. I agree that adding architectures iteratively will likely be simpler due to CI breakages, but this way we can set the precedence.

In addition, I think it would be good to add a warning when the host architecture could not be determined. E.g. something like: `Host architecture could not be determined. Compiling using 'cpu' fallback key which may result in unoptimized code. Please try adding a target.` (probably needs thinking through a bit more) in order to prompt users.